driver/src/contracts/stablex_contract.rs

```diff
@@ -162,17 +162,63 @@ impl StableXContract for StableXContractImpl {
                 .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
         };
 
-        let mut current_block = get_block(BlockNumber::Latest)?;
-        let mut block_number = current_block
+        struct Bounds {
```

Comment:
Can we use https://doc.rust-lang.org/std/ops/struct.Range.html ?

Reply:
It has everything I need except for `.mid()`, but I think it would be a bit of a overhead, especially since I introduced this struct only for readability.

Reply:
I prefer having a `Bounds` struct as well, especially since `Range` is intended as an iterator (it is, in fact, what you create with `1..100`).